src/cli/cli.go

```diff
@@ -121,13 +134,61 @@ func BuildCommandTree(cmd *cobra.Command) {
 	cmd.SetUsageTemplate(indentExamplesTemplate(corsoCmd.UsageTemplate()))
 
 	cmd.CompletionOptions.DisableDefaultCmd = true
+	cmd.SuggestionsMinimumDistance = 2 // default
 
 	repo.AddCommands(cmd)
 	backup.AddCommands(cmd)
 	restore.AddCommands(cmd)
 	export.AddCommands(cmd)
 	debug.AddCommands(cmd)
 	help.AddCommands(cmd)
+	AddCompletion(cmd)
+}
+
+// We are not using the default completion command as it will be
+// harder to control it, for example skipping printing "Logging to file"
+// message
+func AddCompletion(cmd *cobra.Command) {
+	longMessage := `Generate shell completion script for Corso.
+
+These need to be hooked into your shell to enable completions for Corso.
+
+For bash, add the following line to your ` + "`~/.bashrc`  \n" + // two spaces for markdown and \n for go
+		"`eval \"$(corso completion bash)\"`  \n" + `
+
+For zsh, add the following line to your ` + "`~/.zshrc`  \n" +
```

Comment:
I had to do `autoload -U compinit && compinit` prior to running `eval`. Should we also add instructions to add it to zshrc ?